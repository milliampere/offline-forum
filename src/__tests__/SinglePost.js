import React from 'react';
import { mount } from 'enzyme';
import SinglePost from '../components/SinglePost';

describe('<SinglePost />', () => {

  let handleClick;
  let wrapper;

  beforeEach(() => {
    handleClick = jest.fn();
    wrapper = mount(
      <SinglePost 
        title="TestTitle" 
        content="TestContent" 
        id="123" 
        author="TestPerson" 
        currentPersona="TestPerson" 
        date="2018-01-01" 
        onClick={handleClick}
      />);
  });

  it('should display button when author === persona', () => {
    wrapper.setProps({ author: 'OnePerson' });
    wrapper.setProps({ currentPersona: 'OnePerson' });
    expect(wrapper.find('Button').length).toEqual(1);
  })

  it('should not display button when author !== persona', () => {
    wrapper.setProps({ author: 'OnePerson' });
    wrapper.setProps({ currentPersona: 'AnotherPerson' });
    expect(wrapper.find('Button').length).toEqual(0);
  })

  it('should call click with id', () => {
    expect(handleClick).toHaveBeenCalledTimes(0);
    const button = wrapper.find('[data-test="button"]');
    button.simulate('click');
    expect(handleClick).toHaveBeenCalledWith('123');
  }) 

  it('should render a post', () => {
    expect(wrapper).toMatchSnapshot();
  })

})