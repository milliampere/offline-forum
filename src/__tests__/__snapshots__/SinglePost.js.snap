// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SinglePost /> should render the post 1`] = `
<SinglePost
  author="TestPerson"
  content="TestContent"
  currentPersona="TestPerson"
  date="2018-01-01"
  id="123"
  onClick={[Function]}
  title="TestTitle"
>
  <div
    className="w-full shadow p-6 m-6 border rounded relative"
  >
    <article
      className="mb-2"
    >
      <Button
        className="absolute pin-t pin-r mx-4 my-6"
        danger={true}
        onClick={[Function]}
      >
        <button
          className="bg-red-dark hover:bg-red-darker text-white font-bold py-2 px-4 rounded-full absolute pin-t pin-r mx-4 my-6"
          data-test="button"
          onClick={[Function]}
        >
          x
        </button>
      </Button>
      <h2
        className="mb-4 w-4/5"
      >
        TestTitle
      </h2>
      <ReactMarkdown
        className="mb-4 text-grey-darkest markdown"
        escapeHtml={true}
        renderers={Object {}}
        skipHtml={false}
        source="TestContent"
        transformLinkUri={[Function]}
      >
        <div
          className="mb-4 text-grey-darkest markdown"
          key="root-1-1"
        >
          <p
            key="paragraph-1-1"
          >
            TestContent
          </p>
        </div>
      </ReactMarkdown>
      <p
        className="text-grey-dark mb-4"
      >
        Posted by: 
        TestPerson
         @ 
        2018-01-01
      </p>
    </article>
    <Comments
      currentPersona="TestPerson"
      postId="123"
    >
      <div
        className="py-2"
      >
        <h2
          className="text-indigo-darker border-b mb-4"
        >
          Comments
        </h2>
        <CreateNewComment
          author="TestPerson"
          postId="123"
          updateComments={[Function]}
        >
          <form
            className="container mx-auto flex flex-col p-6"
            onSubmit={[Function]}
          >
            <label
              className="mb-2"
              htmlFor="comment"
            >
              Comment
            </label>
            <textarea
              className="shadow rounded border mb-2 p-2"
              id="comment"
              name="comment"
              onChange={[Function]}
              value=""
            />
            <input
              className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
              type="submit"
              value="Comment"
            />
          </form>
        </CreateNewComment>
      </div>
    </Comments>
  </div>
</SinglePost>
`;
